aiida_common_workflows/workflows/dissociation.py

```diff
@@ -0,0 +1,216 @@
+        spec.input('sub_process_class', non_db=True, validator=validate_sub_process_class)
+        spec.inputs.validator = validate_inputs
+        spec.outline(
+            cls.run_init,
+            cls.run_dissociation,
+            cls.inspect_results,
+        )
+        spec.output('curve_data', valid_type=orm.ArrayData,
+            help='The computed total energy (and eventually spin) versus distance.')
+        spec.exit_code(400, 'ERROR_SUB_PROCESS_FAILED',
+            message='At least one of the `{cls}` sub processes did not finish successfully.')
+
+    def get_distances(self):
+        """Return the list of scale factors."""
+        if 'distances' in self.inputs:
+            return self.inputs.distances
+
+        count = self.inputs.distances_count.value
+        maxim = self.inputs.distance_max.value
+        minim = self.inputs.distance_min.value
+        return [orm.Float(minim + i * (maxim-minim) / (count-1)) for i in range(count)]
+
+    def get_sub_workchain_builder(self, distance, previous_workchain=None):
+        """Return the builder for the relax workchain."""
+        molecule = set_distance(self.inputs.molecule, distance)
+        process_class = WorkflowFactory(self.inputs.sub_process_class)
+
+        builder = process_class.get_inputs_generator().get_builder(
+            molecule,
+            previous_workchain=previous_workchain,
+            **self.inputs.generator_inputs
+        )
+        builder._update(**self.inputs.get('sub_process', {}))  # pylint: disable=protected-access
+
+        return builder
+
+    def run_init(self):
+        """Run the first workchain."""
+        distance = self.get_distances()[0]
+        builder = self.get_sub_workchain_builder(distance)
+        self.report(f'submitting `{builder.process_class.__name__}` for distance `{distance.value}`')
+        self.ctx.previous_workchain = self.submit(builder)
+        self.to_context(children=append_(self.ctx.previous_workchain))
+
+    def run_dissociation(self):
+        """Run the sub process at each distance to compute the total energy."""
+        for distance in self.get_distances()[1:]:
+            previous_workchain = self.ctx.previous_workchain
+            builder = self.get_sub_workchain_builder(distance, previous_workchain=previous_workchain)
+            self.report(f'submitting `{builder.process_class.__name__}` for dinstance `{distance.value}`')
+            self.to_context(children=append_(self.submit(builder)))
+
+    def inspect_results(self):
+        """
+        Inspect all children workflows to make sure they finished successfully.
+        Collect the total energies and return an array with the results.
+        """
+        if any([not child.is_finished_ok for child in self.ctx.children]):
+            return self.exit_codes.ERROR_SUB_PROCESS_FAILED.format(cls=self.inputs.sub_process_class)  # pylint: disable=no-member
+
+        self.report('distance (ang),  energy (eV)')
```

Comment:
Do we really need to report the values here? Is it useful to have in the log?

Reply:
We do it for the eos, so why not here?